server/datastore/mysql/activities.go

```diff
@@ -231,14 +231,31 @@ func (ds *Datastore) MarkActivitiesAsStreamed(ctx context.Context, activityIDs [
 // software to install, etc.) and provides a unified view of those upcoming
 // tasks.
 func (ds *Datastore) ListHostUpcomingActivities(ctx context.Context, hostID uint, opt fleet.ListOptions) ([]*fleet.Activity, *fleet.PaginationMetadata, error) {
+	// NOTE: Be sure to update both the count (here) and list statements (below)
+	// if the query condition is modified.
 	countStmts := []string{
-		`SELECT COUNT(*) c FROM host_script_results WHERE host_id = :host_id AND exit_code IS NULL`,
-		`SELECT COUNT(*) c FROM host_software_installs WHERE host_id = :host_id AND pre_install_query_output IS NULL AND install_script_exit_code IS NULL`,
+		`SELECT
+			COUNT(*) c
+			FROM host_script_results
+			WHERE host_id = :host_id AND
+						exit_code IS NULL AND
+						(sync_request = 0 OR created_at >= DATE_SUB(NOW(), INTERVAL :max_wait_time SECOND))`,
+		`SELECT
+			COUNT(*) c
+			FROM host_software_installs
+			WHERE host_id = :host_id AND
+						pre_install_query_output IS NULL AND
+						install_script_exit_code IS NULL`,
 	}
 
 	var count uint
 	countStmt := `SELECT SUM(c) FROM ( ` + strings.Join(countStmts, " UNION ALL ") + ` ) AS counts`
-	countStmt, args, err := sqlx.Named(countStmt, map[string]any{"host_id": hostID})
+
+	seconds := int(scripts.MaxServerWaitTime.Seconds())
```

Comment:
this was here before so no action needed, and it probably doesn't make much of a difference but, are we using the right var? 

https://github.com/fleetdm/fleet/blob/1f6534f93395987c0f3e635641fc54b179de756f/pkg/scripts/scripts.go#L7-L16

Reply:
That's a good question... Thinking about it, I _think_ we do, because we need a bit of wiggle room to allow orbit to ping the fleet server and receive the notifications, so it's a bit more than the exact time to run? I could see it being ok with either of those values, I believe the important thing is that we use the same in the list upcoming activities and in list pending script executions (the datastore method used to send scripts to execute to fleetd).